GoodMorning/GoodMorning/Domain/UseCases/FetchCurrentWeatherUseCase.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  FetchCurrentWeatherUseCase.swift
+//  GoodMorning
+//
+//  Created by Miro on 11/2/23.
+//
+
+import Foundation
+
+final class FetchCurrentWeatherUseCase {
+
+    private let weatherRepository: WeatherRepository
+    private let locationRepository: CurrentLocationRepository
+
+    init(weatherRepository: WeatherRepository, locationRepository: CurrentLocationRepository) {
+        self.weatherRepository = weatherRepository
+        self.locationRepository = locationRepository
+    }
+
+    func execute() async throws -> CurrentWeather {
+
+        let locationResult = locationRepository.fetchCurrentLocation()
+
+        switch locationResult {
+        case .success(let coordinate):
+            let currentWeatherResult = try await weatherRepository.fetchCurrentWeather(
+                in: coordinate ?? Coordinate(longitude: "0", latitude: "0")
+            )
+
+            switch currentWeatherResult {
+            case .success(let weather):
+                return weather
+            case .failure(let networkError):
+                throw networkError
+            }
+
+        case .failure(let error):
+            throw error
+        case .none:
+            throw LocationError.unknowned
```

Comment:
미로의 PR 설명을 보니 locationResult가 옵셔널 타입이기 때문에 switch문을 사용할 때 옵셔널이 nil인 상황을 `case .none` 으로 표현한 것이 조금 걸리신다고 이해했어요.

혹시 이 부분을
```suggestion
guard let locationResult = locationRepository.fetchCurrentLocation() else {
        throw LocationError.unknowned
}
```
이렇게 변경해서 `case .none`을 삭제하는건 어떨까요?!!

Reply:
앗 저는 `.none` 자체를 사용하는 것의 문제라기 보다는 애초에 옵셔널일 수 밖에 없는 방식으로 코드를 짠 것이 조금 걸린다는 말이었습니다🥲(Swift concurrency와 DispatchQueue를 같이 사용해서 생기는 문제점) 해당 부분은 디코로 한번 이야기해보면 좋을 거 같아요!!ㅎㅎ